### ISSUE TITLE
A11y: section page content

### ISSUE THREAD

**Issue by jennur** [inveniosoftware/invenio-search-ui#182]
The filters and search results should be sectioned to make screen reader navigation easier.
- Replace surrounding `<div>` with `<section aria-label="section-title">`
- 
![Screenshot 2023-09-04 at 10 35 18](https://github.com/inveniosoftware/invenio-search-ui/assets/21052053/0765acd8-de6f-4785-b732-b9cd601a4950)
